### ISSUE TITLE
Question: validation_ds for En Conformer-CTC Small LibriSpeech configuration.

### ISSUE THREAD

**Issue by june-oh** [NVIDIA/NeMo#3617]
this is a question about the pre-trained [model](https://catalog.ngc.nvidia.com/orgs/nvidia/teams/nemo/models/stt_en_conformer_ctc_small_ls).

I downloaded the model and used the following command to check the training-related conf.

```
import nemo.collections.asr as nemo_asr
asr_model = nemo_asr.models.EncDecCTCModelBPE.from_pretrained(model_name="<path_of_pretrained_model>")
asr_model._cfg
```

And I checked the following.
>{'sample_rate': 16000, 'log_prediction': True, 'ctc_reduction': 'mean_batch', 'train_ds': {'manifest_filepath': '/data/Librispeech_SP_Tarred/tarred_audio_manifest.json', 'sample_rate': 16000, 'batch_size': 32, 'shuffle': True, 'num_workers': 8, 'pin_memory': True, 'use_start_end_token': False, 'trim_silence': False, 'max_duration': 16.7, 'min_duration': 0.1, 'shuffle_n': 2048, 'is_tarred': True, 'tarred_audio_filepaths': '/data/Librispeech_SP_Tarred/audio__OP_0..511_CL_.tar'}, 'validation_ds': {'manifest_filepath': ['/manifests/librispeech/librivox-dev-other.json', '/manifests/librispeech/librivox-dev-clean.json', '/manifests/librispeech/librivox-test-other.json', '/manifests/librispeech/librivox-test-clean.json'], 'sample_rate': 16000, 'batch_size': 32, 'shuffle': False, 'num_workers': 8, 'pin_memory': True, 'use_start_end_token': False, 'is_tarred': False, 'tarred_audio_filepaths': '/data/LibriSpeech/eval__OP_0..1023_CL_.tar'}, 'test_ds': {'manifest_filepath': ['/manifests/librispeech/librivox-dev-other.json', '/manifests/librispeech/librivox-dev-clean.json', '/manifests/librispeech/librivox-test-other.json', '/manifests/librispeech/librivox-test-clean.json'], 'sample_rate': 16000, 'batch_size': 32, 'shuffle': False, 'num_workers': 8, 'pin_memory': True, 'use_start_end_token': False, 'is_tarred': False, 'tarred_audio_filepaths': '/data/LibriSpeech/eval__OP_0..1023_CL_.tar'}, 'tokenizer': {'dir': '/tokenizers/librispeech/experimental/tokenizer_spe_unigram_v128_max0/', 'type': 'bpe'}, 'preprocessor': {'_target_': 'nemo.collections.asr.modules.AudioToMelSpectrogramPreprocessor', 'sample_rate': 16000, 'normalize': 'per_feature', 'window_size': 0.025, 'window_stride': 0.01, 'window': 'hann', 'features': 80, 'n_fft': 512, 'log': True, 'frame_splicing': 1, 'dither': 1e-05, 'pad_to': 0, 'pad_value': 0.0}, 'spec_augment': {'_target_': 'nemo.collections.asr.modules.SpectrogramAugmentation', 'freq_masks': 2, 'time_masks': 5, 'freq_width': 27, 'time_width': 0.05}, 'encoder': {'_target_': 'nemo.collections.asr.modules.ConformerEncoder', 'feat_in': 80, 'feat_out': -1, 'n_layers': 16, 'd_model': 176, 'subsampling': 'striding', 'subsampling_factor': 4, 'subsampling_conv_channels': 176, 'ff_expansion_factor': 4, 'self_attention_model': 'rel_pos', 'n_heads': 4, 'att_context_size': [-1, -1], 'xscaling': True, 'untie_biases': True, 'pos_emb_max_len': 5000, 'conv_kernel_size': 31, 'dropout': 0.1, 'dropout_emb': 0.0, 'dropout_att': 0.1}, 'decoder': {'_target_': 'nemo.collections.asr.modules.ConvASRDecoder', 'feat_in': 176, 'num_classes': 128, 'vocabulary': ['<unk>', 'e', 's', '▁', 't', 'a', 'o', 'i', '▁the', 'd', 'l', 'n', '▁a', 'm', 'y', 'u', '▁s', 'p', 'ed', 'c', '▁and', 're', '▁to', '▁of', 'r', 'w', 'ing', '▁w', 'h', '▁p', '▁c', 'er', 'f', 'k', 'ar', '▁in', '▁f', '▁b', 'g', 'an', 'in', '▁i', 'en', '▁he', 'le', '▁g', 'or', 'll', 'b', '▁be', 'ro', 'st', 'on', '▁d', 'v', 'ly', 'ce', 'ur', 'es', '▁that', '▁o', 'us', '▁was', '▁it', '▁th', 've', 'ch', 'un', 'al', '▁t', '▁ma', 'ri', '▁you', '▁on', 'ver', 'ent', '▁for', '▁re', 'ra', "'", '▁his', 'ir', 'ter', '▁with', '▁her', 'it', 'th', '▁mo', '▁me', '▁ha', '▁e', '▁as', 'tion', '▁had', '▁not', '▁no', '▁do', 'ther', '▁but', '▁st', '▁she', '▁is', 'igh', '▁ho', '▁lo', 'ng', '▁him', '▁an', 'ck', 'j', 'ugh', '▁de', '▁li', '▁mi', '▁la', '▁my', '▁con', '▁have', '▁this', '▁which', 'q', '▁up', '▁said', '▁from', '▁who', '▁ex', 'x', 'z']}, 'optim': {'name': 'adamw', 'lr': 5.0, 'betas': [0.9, 0.98], 'weight_decay': 0.001, 'sched': {'name': 'NoamAnnealing', 'd_model': 176, 'warmup_steps': 10000, 'warmup_ratio': None, 'min_lr': 1e-06}}, 'target': 'nemo.collections.asr.models.ctc_bpe_models.EncDecCTCModelBPE'},

I paid attention to the next part.

>'validation_ds': {'manifest_filepath': ['/manifests/librispeech/librivox-dev-other.json', '/manifests/librispeech/librivox-dev-clean.json', '/manifests/librispeech/librivox-test-other.json', '/manifests/librispeech/librivox-test-clean.json'],

Did you use data other than LS-dev for LS training?
Or, is it a default value of the model not used in real training?

**Comment by titu1994** [comment]
Seems that checkpoint was trained such that it will plot the scores of all 4 datasets, since evaluating evaluating all of the datasets after training for the hundreds of models we try would be tedious. 

However, Nemo will only use the 0th dataset for decisions such as selection of checkpoint and weight saving etc (in this case it's dev other)

**Comment by titu1994** [comment]
We don't use the validation and test_ds during training, only train_ds part of the config. The validation_ds is used for evaluating the model performance during training itself to see if the model is converging or not.

**Comment by june-oh** [comment]
@titu1994 
Thank you for the reply.
I got it.
